src/goLiveErrors.ts

```diff
@@ -62,26 +64,33 @@ function processFile(e: vscode.TextDocumentChangeEvent) {
 			return;
 		}
 
-		errorDiagnosticCollection.delete(uri);
+		errorDiagnosticCollection.clear();
 
 		if (err) {
 			// we want to take the error path here because the command we are calling
 			// returns a non-zero exit status if the checks fail
-			let diagnostics = [];
+			let diagnosticMap: Map<string, vscode.Diagnostic[]> = new Map();
 
 			stderr.split('\n').forEach(error => {
 				if (error === null || error.length === 0) {
 					return;
 				}
 				// extract the line, column and error message from the gotype output
-				let [_, line, column, message] = /^.+:(\d+):(\d+):\s+(.+)/.exec(error);
-
+				let [_, file, line, column, message] = /^(.+):(\d+):(\d+):\s+(.+)/.exec(error);
 				let range = new vscode.Range(+line - 1, +column, +line - 1, +column);
 				let diagnostic = new vscode.Diagnostic(range, message, vscode.DiagnosticSeverity.Error);
+
+				let diagnostics = diagnosticMap.get(file);
+				if (!diagnostics) {
+					diagnostics = [];
+				}
 				diagnostics.push(diagnostic);
+				diagnosticMap.set(file, diagnostics);
 			});
 
-			errorDiagnosticCollection.set(uri, diagnostics);
+			diagnosticMap.forEach((diagnostics, file) => {
+				errorDiagnosticCollection.set(vscode.Uri.parse('file://' + file), diagnostics);
```

Comment:
just curious... why add the prefix `file://` before parsing the file to get the uri? `vscode.Uri.parse` should work with just the file path as well